lib/lambda_ethereum_consensus/state_transition/operations.ex

```diff
@@ -0,0 +1,189 @@
+defmodule LambdaEthereumConsensus.StateTransition.Operations do
+  @moduledoc """
+  State transition Operations related functions
+  """
+
+  alias LambdaEthereumConsensus.StateTransition.Accessors
+  alias LambdaEthereumConsensus.StateTransition.Mutators
+  alias LambdaEthereumConsensus.StateTransition.Predicates
+  alias SszTypes.BeaconState
+  alias SszTypes.ExecutionPayload
+  alias SszTypes.Validator
+  alias SszTypes.Withdrawal
+
+  @doc """
+  Apply withdrawals to the state.
+  """
+  @spec process_withdrawals(BeaconState.t(), ExecutionPayload.t()) ::
+          {:ok, BeaconState.t()} | {:error, String.t()}
+  def process_withdrawals(
+        %BeaconState{
+          validators: validators,
+          next_withdrawal_validator_index: next_withdrawal_validator_index
+        } = state,
+        %ExecutionPayload{withdrawals: withdrawals}
+      ) do
+    expected_withdrawals = get_expected_withdrawals(state)
+    result = decrease_balances(state, withdrawals, expected_withdrawals)
+
+    response =
+      case result do
```

Comment:
Instead of 

```elixir
case decrease_balances(state, withdrawals, expected_withdrawals) do
  {:ok, state} -> do_something(state)
  {:error, error} -> {:error, error}
end
```

we can just:

```elixir

with {:ok, state} <- decrease_balances(state, withdrawals, expected_withdrawals) do
   do_something(state)
end
```

`with` defaults to whatever was returned if it doesn't match. We're also missing the length check before deducing the balances, so we should probably go:

```elixir
lengths_check = if length(withdrawals) == length(expected_withdrawals) do
  :ok
else
  {:error, "expected withdrawals don't match the state withdrawals in length"}
end

with :ok <- lengths_check,
     {:ok, state} <- decrease_balances(state, withdrawals, expected_withdrawals) do
  do_something(state)
end
```